File: src/main/kotlin/at/fhhgb/mc/component/sim/model/GeoDoublePoint.java
package at.fhhgb.mc.component.sim.model;

/**
 * A point representation which calculates its position with polar coordinates.
 *
 * @author Michael Nigl
 */
public class GeoDoublePoint {

    private double m_x = 0.0;
    private double m_y = 0.0;


    /**
     * The constructor which initializes the coordinates of the point.
     *
     * @param _x - x coordinate
     * @param _y - y coordinate
     */
    public GeoDoublePoint(double _x, double _y) {
        m_x = _x;
        m_y = _y;
    }

    /**
     * Gives the x coordinate of the point.
     *
     * @return - value if the x coordinate.
     */
    public double getX() {
        return m_x;
    }

    /**
     * Gives the y coordinate of the point.
     *
     * @return - value if the y coordinate.
     */
    public double getY() {
        return m_y;
    }

    /**
     * Calculates the distance from the center of the coordinate system to the point.
     * (Polar coordinate)
     *
     * @return - the distance of the point
     */
    public double length() {
        return Math.sqrt(Math.pow(m_x, 2) + Math.pow(m_y, 2));
    }

    /**
     * Prints a string representation of this point with its coordinates.
     *
     * @return the string of this point
     */
    public String toString() {
        return "Point Coordinates: x: " + m_x + " y: " + m_y;
    }

}

File: src/main/kotlin/at/fhooe/mc/wifipositioning/model/graphics/GeoDoublePoint.kt
package at.fhooe.mc.wifipositioning.model.graphics

/**
 * A point representation which calculates its position with polar coordinates.
 */
class GeoDoublePoint(var x: Double, var y: Double) {

    /**
     * Calculates the distance from the center of the coordinate system to the point.
     * (Polar coordinate)
     *
     * @return - the distance of the point
     */
    fun length(): Double {
        return Math.sqrt(Math.pow(x, 2.0) + Math.pow(y, 2.0))
    }

    /**
     * Prints a string representation of this point with its coordinates.
     *
     * @return the string of this point
     */
    override fun toString(): String {
        return "Point Coordinates: x: $x y: $y"
    }

}
